packages/config/src/discovery/ProjectDiscovery.ts

```diff
@@ -981,16 +1015,37 @@ export function stringFormat(str: string, ...val: string[]) {
   return str
 }
 
-const roleDescriptions: { [key in StackRole]: string } = {
-  Sequencer:
-    'Sequencer is an actor allowed to commit transactions from current layer to the host chain.',
-  Proposer:
-    'Proposer is an actor allowed to post new state roots of current layer to the host chain.',
-  Challenger:
-    'Challenger is an actor allowed to delete state roots proposed by a Proposer.',
-  Guardian: 'Guardian is an actor allowed to pause deposits and withdrawals.',
-  Validator:
-    'Validator is an actor that validates the correctness of state transitions.',
+const roleDescriptions: {
+  [key in (typeof RolePermissionEntries)[number]]: {
```

Comment:
Can we use description of the permission as defined in the template?

Reply:
just a note: Validator role is only temporary and will be removed later exactly because of this ambiguity between the validators in Arbitrum and validators of say ethereum or any other pos network

Reply:
It's going to duplicate the code even more. Since we're repeating the permission validator in many different places since most of the code is not yet templatized that well.